ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactHorizontalScrollView.java

```diff
@@ -537,6 +541,51 @@ protected void onOverScrolled(int scrollX, int scrollY, boolean clampedX, boolea
     super.onOverScrolled(scrollX, scrollY, clampedX, clampedY);
   }
 
+  private int getMaxScrollX() {
+    int contentWidth = mContentView.getWidth();
+    int viewportWidth = getWidth() - getPaddingLeft() - getPaddingRight();
+    return Math.max(0, contentWidth - viewportWidth);
+  }
+
+  @Override
+  public void onChildViewAdded(View parent, View child) {
+    mContentView = child;
+    mContentView.addOnLayoutChangeListener(this);
+  }
+
+  @Override
+  public void onChildViewRemoved(View parent, View child) {
+    mContentView.removeOnLayoutChangeListener(this);
+    mContentView = null;
+  }
+
+  /**
+   * Called when a mContentView's layout has changed. Fixes the scroll position if it's too large
+   * after the content resizes. Without this, the user would see a blank ScrollView when the scroll
+   * position is larger than the ScrollView's max scroll position after the content shrinks.
+   */
+  @Override
+  public void onLayoutChange(
```

Comment:
Would setting `fillViewport` achieve the same behavior? If Android already handles it, the manually calculations may not needed. 

https://developer.android.com/reference/android/widget/HorizontalScrollView.html#setFillViewport(boolean)

Reply:
`fillViewPort` will change how the child views are being measured, typically it might result in different behaviour in a case where content size is less than viewport size.
I will try to come up with a case where fillViewport might break existing behaviours.

Reply:
Even with fillViewPort set to true, the issue does not go away.
<img width="940" alt="Screen Shot 2020-02-16 at 12 20 10 PM" src="https://user-images.githubusercontent.com/659202/74600378-cd7cce80-50b6-11ea-93c5-536174f1de98.png">

hence the original solution